### DISCUSSION TITLE
Is it possible to minify Baby Buddy's javascript?

### DISCUSSION THREAD

**Discussion by gbballpack** [babybuddy/babybuddy#491] [Ideas]
Using Cloudflare with Baby Buddy, Baby Buddy running inside of Docker with an external reverse proxy, Cloudflare is able to minify about 80 KiB from the vendor JS file.

Would it be possible to have everyone run a minified vendor JS file? Minified files generally run quicker and can reduce power consumption when on mobile compared to non-minified JS files.

I am curious if it is possible to share the benefits of the minified JS with those not using Cloudflare?

Before and after screenshots are attached, using https://web.dev/measure to verify the Cloudflare JS minification.
![IMG_1425](https://user-images.githubusercontent.com/39963341/176254529-3fe168c2-6552-4fcb-b0b9-c8ad65b039a3.jpg)
![IMG_1426](https://user-images.githubusercontent.com/39963341/176254561-8fdb094a-c5db-4025-ab60-f67a0a91351d.jpg)

**Comment by cdubz** [comment]
Hm yeah it looks CF is stripping something extra out of the combined vendor JS. I haven’t reviewed the static files tooling in ages so there are probably advancements there we are not taking advantage of.

I’m gonna convert this to an issue — worth looking at closer.

**Comment by gbballpack** [comment]
For a past project based on PHP/nodejs I used UglifyJS to compress down the static JS files. An example of UglifyJS's compression can be found here: [https://www.keycdn.com/support/uglifyjs](https://www.keycdn.com/support/uglifyjs)

In researching something similar for Django, it appears that [Django Compressor](https://github.com/django-compressor/django-compressor) supports an UglifyJS wrapper and a number of JS compressors. At a glance, Django Compressor appears to be a promising tool to compress javascript. I wish I could write a PR for this but I have zero experience working with frameworks like Django and little time to learn Django, currently.
